src/states/Preroll.js

```diff
@@ -21,28 +21,33 @@ export default class Preroll extends AdState {
    * happen here, not in a constructor.
    */
   init(player, adsReady, shouldResumeToContent) {
-    this.waitingForAdBreak = true;
 
-    // Loading spinner from now until ad start or end of ad break.
-    player.addClass('vjs-ad-loading');
+    // With stitched ads, we won't need to wait for a preroll and we can
+    // allow the integration to call startLinearAdMode whenever it's ready.
+    if (!player.ads.stitchedAds()) {
+      this.waitingForAdBreak = true;
```

Comment:
Technically we're still waiting for ad break until startLinearAdMode, even though with stitched ads this will be a short time

Reply:
I can see skipping the loading spinner, I wonder if nopreroll should be handled?

Reply:
In our SSAI plugin, at least, we do fire it if there's no preroll stitched in. Makes sense to skip the spinner and preroll timeout only probably. I was thinking we might want the `resumeAfterNoPreroll` below.

Reply:
👍